Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Middlewares/RpcV2CborValidateResponseHeaderMiddleware.swift

```diff
@@ -0,0 +1,45 @@
+//
```

Comment:
Why does this only check `smithy-protocol` header and not `Content-Type`, `Content-Length` and `x-amzn-ErrorType` headers? Not suggesting they need to be checked, but rather asking why only one header was singled out to require verification on SDK side

Reply:
This section of the SEP:

```
Operation response deserialization
Clients MUST use the following rules to interpret responses and determine how to deserialize them:

1. If the response does not have the same smithy-protocol header as the request, it MUST be considered malformed. Clients who receive a malformed response MUST handle it (i.e. throw a reasonable exception) based solely on the HTTP response code. No attempt should be made to interpret the response body or headers.
2. If the response code is 200, the request is successful and the response payload SHALL be deserialized as the output shape defined on the operation.
3. Finally, if the response code is not 200, the response payload is an exception and SHALL be deserialized according to [Operation error deserialization](https://code.amazon.com/packages/AwsDrSeps/blobs/main/--/seps/accepted/shared/rpcv2-cbor/protocol-selection_smithy-rpc-v2.md#operation-error-deserialization).
```

This middleware validates only smithy-protocol header and continues on to deserialize the request if and only if its correctly set. I did not check other headers due to the part "No attempt should be made to interpret the response body or headers." which I assumed to mean "other" headers.